### PR TITLE
feat(events): implement update-schedule-item event handler

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal
Implement the handler for `update-scheduled-item`

## I'd love feedback/perspectives on:
- Anything
- Is this the intended behavior per the ticket?

## Implementation Decisions
- Similar to add-scheduled-item

## References

JIRA ticket: INFRA-352

### CODE REVIEW

File: curation-migration-datasync/database/dataService.ts [hunk 1]
```diff
@@ -130,6 +130,70 @@ export class DataService {
...
+    eventBody: ScheduledItemPayload,
+    curatedRecId: number,
+    resolvedId: number,
+    domainId: string
+  ) {
+    const topicId = await this.getTopicIdByName(
+      getTopicForReaditLaTmpDatabase(eventBody.topic)
+    );
+
+    const trx = await this.db.transaction();
```
Comment (kschelonka) [inline curation-migration-datasync/database/dataService.ts:150]: If you use the transaction handler, you don't have to call commit and rollback (it does this automatically). I have some updated handler function code to do error reporting, so we don't need to put all the event data in this error anymore.

File: curation-migration-datasync/database/dataService.ts [hunk 2]
```diff
@@ -130,6 +130,70 @@ export class DataService {
...
+      await trx(config.tables.curatedFeedItems)
+        .update(curatedItem)
+        .where({ curated_rec_id: curatedRecId });
+
+      await trx.commit();
+
+      return curatedRecId;
+    } catch (e) {
+      await trx.rollback();
+      throw new Error(
```
Comment (kschelonka) [inline curation-migration-datasync/database/dataService.ts:187]: I have some updated handler function code in #116 to do error reporting, so we don't need to put all the event data in this error anymore.

**Comment by kschelonka** [comment]
Don't forget to call your new function in the handler index.ts

File: curation-migration-datasync/database/dataService.ts (hunk 1 above)
Comment (kkyeboah) [reply curation-migration-datasync/database/dataService.ts:150]: did this in order to throw a specific error, I can update if that is not a requirement.

File: curation-migration-datasync/eventConsumer.ts [hunk 3]
```diff
@@ -55,3 +57,38 @@ export async function removeScheduledItem(
...
+  const dbService = new DataService(db);
+  const parserResponse = await getParserMetadata(eventBody.url);
+
+  await dbService.updateScheduledItem(
+    eventBody,
+    scheduledItem.curatedRecId,
+    parseInt(parserResponse.resolvedId),
+    parserResponse.domainId
+  );
+}
```
Comment (sri-kirsh) [inline curation-migration-datasync/eventConsumer.ts:94]: we have to update the dynamo record's lastUpdatedAt.

i have something like this:
 ```
curatedItem.lastUpdatedAt = Math.round(new Date().getTime() / 1000);
await curatedItemModel.insert(curatedItem);
```

File: curation-migration-datasync/eventConsumer.ts [hunk 4]
```diff
@@ -55,3 +57,38 @@ export async function removeScheduledItem(
...
+  const curatedItemModel = new CuratedItemRecordModel();
+  const scheduledItem = await curatedItemModel.getByScheduledItemExternalId(
+    eventBody.scheduledItemExternalId
+  );
+
+  if (!scheduledItem) {
+    const errorMessage = `update-scheduled-item: No mapping found for scheduledItemExternalId=${eventBody.scheduledItemExternalId}`;
+    Sentry.captureMessage(errorMessage);
+    console.log(errorMessage);
+    await addScheduledItem(eventBody, db);
```
Comment (sri-kirsh) [inline curation-migration-datasync/eventConsumer.ts:81]: nice! coz the `scheduledItemEventPayload` gets all the required fields, https://github.com/Pocket/curated-corpus-api/blob/main/src/events/eventBus/EventBusConfig.ts#L58-L59

incase if the scheduledItem is missing in dynamo, we can just treat it as a new item and add them.

**Review by kschelonka — APPROVED** [review]

**Pull request merged**